beacon-chain/sync/validate_beacon_attestation.go

```diff
@@ -327,21 +327,49 @@ func validateAttestingIndex(
 }
 
 // Returns true if the attestation was already seen for the participating validator for the slot.
-func (s *Service) hasSeenCommitteeIndicesSlot(slot primitives.Slot, committeeID primitives.CommitteeIndex, aggregateBits []byte) bool {
+func (s *Service) hasSeenUnaggregatedAtt(att eth.Att) bool {
 	s.seenUnAggregatedAttestationLock.RLock()
 	defer s.seenUnAggregatedAttestationLock.RUnlock()
-	b := append(bytesutil.Bytes32(uint64(slot)), bytesutil.Bytes32(uint64(committeeID))...)
-	b = append(b, aggregateBits...)
+	var attester uint64
+	if att.Version() >= version.Electra {
+		if !att.IsSingle() {
+			log.Debug("Called hasSeenUnaggregatedAtt with a non-single Electra attestation")
+		}
+		attester = uint64(att.GetAttestingIndex())
+	} else {
+		aggBits := att.GetAggregationBits()
+		if aggBits.Count() != 1 {
+			log.Debug("Attestation does not have exactly 1 bit set. It will be considered not seen")
+			return false
+		}
+		attester = uint64(att.GetAggregationBits().BitIndices()[0])
+	}
+	b := append(bytesutil.Bytes32(uint64(att.GetData().Slot)), bytesutil.Bytes32(uint64(att.GetCommitteeIndex()))...)
+	b = append(b, bytesutil.Bytes32(attester)...)
 	_, seen := s.seenUnAggregatedAttestationCache.Get(string(b))
 	return seen
 }
 
-// Set committee's indices and slot as seen for incoming attestations.
-func (s *Service) setSeenCommitteeIndicesSlot(slot primitives.Slot, committeeID primitives.CommitteeIndex, aggregateBits []byte) {
+// Set an incoming attestation as seen for the participating validator for the slot.
+func (s *Service) setSeenUnaggregatedAtt(att eth.Att) {
 	s.seenUnAggregatedAttestationLock.Lock()
 	defer s.seenUnAggregatedAttestationLock.Unlock()
-	b := append(bytesutil.Bytes32(uint64(slot)), bytesutil.Bytes32(uint64(committeeID))...)
-	b = append(b, bytesutil.SafeCopyBytes(aggregateBits)...)
+	var attester uint64
+	if att.Version() >= version.Electra {
+		if !att.IsSingle() {
+			log.Debug("Called setSeenUnaggregatedAtt with a non-single Electra attestation")
```

Comment:
Why wouldn't we return here? How is this different than say `aggBits.Count() != 1`

Reply:
Missed this too, thank you.